The attached data file (acquired via Kaggle) contains information about historical video game sales, including the rank (in sales), title, platform (system), release year, genre, publisher, and sales (in millions) in North America, Europe, Japan, other countries, and in total. Complete the following exercises using the data.

In [10]:
import pandas as pd

### Exercise #1 

Load the data file into a data frame, setting the Rank column as the index.

In [4]:
data = pd.read_csv('vgsales.csv', index_col='Rank')

Preview the first ten titles in the data set.

In [5]:
data.head(10)

Name Platform    Year         Genre Publisher  \
Rank                                                                       
1                    Wii Sports      Wii  2006.0        Sports  Nintendo   
2             Super Mario Bros.      NES  1985.0      Platform  Nintendo   
3                Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
4             Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
5      Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   
6                        Tetris       GB  1989.0        Puzzle  Nintendo   
7         New Super Mario Bros.       DS  2006.0      Platform  Nintendo   
8                      Wii Play      Wii  2006.0          Misc  Nintendo   
9     New Super Mario Bros. Wii      Wii  2009.0      Platform  Nintendo   
10                    Duck Hunt      NES  1984.0       Shooter  Nintendo   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                           
1        41.49     29.02      3.77         8.46         82.74  
2        29.08      3.58      6.81         0.77         40.24  
3        15.85     12.88      3.79         3.31         35.82  
4        15.75     11.01      3.28         2.96         33.00  
5        11.27      8.89     10.22         1.00         31.37  
6        23.20      2.26      4.22         0.58         30.26  
7        11.38      9.23      6.50         2.90         30.01  
8        14.03      9.20      2.93         2.85         29.02  
9        14.59      7.06      4.70         2.26         28.62  
10       26.93      0.63      0.28         0.47         28.31

Report the total number of titles.

In [6]:
len(data['Name'])

16598

Some summary statistics about the distribution of global sales.

In [7]:
data['Global_Sales'].describe()

count    16598.000000
mean         0.537441
std          1.555028
min          0.010000
25%          0.060000
50%          0.170000
75%          0.470000
max         82.740000
Name: Global_Sales, dtype: float64

What proportion (or percentage) of titles earned at least $1 million in global sales?

In [8]:
(len(data['Name'][data['Global_Sales'] >= 1])/len(data['Name']))*100

12.537655139173395

### Exercise #2

Write a function top_sellers that takes in the DataFrame of video game sales (df), a category (cat, default=None), value (val, default=None), and a number (n, default=5) that specifies the number of rows to display in the result. The function should return a DataFrame containing the top n video games (in terms of global sales) for a specific value in the category (if specified) or the top n video games overall (if cat and val are not specified).

In [42]:
def top_sellers(df=data, cat=None, val=None, n=5):
    if cat != None and val != None:
        return data[data[cat]==val].head(n) 
    else : 
        return data.head(n)

Use the function to display the following top selling games:

Top 5 overall sellers

In [43]:
top_sellers(n=5)

Name Platform    Year         Genre Publisher  \
Rank                                                                      
1                   Wii Sports      Wii  2006.0        Sports  Nintendo   
2            Super Mario Bros.      NES  1985.0      Platform  Nintendo   
3               Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
4            Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
5     Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                           
1        41.49     29.02      3.77         8.46         82.74  
2        29.08      3.58      6.81         0.77         40.24  
3        15.85     12.88      3.79         3.31         35.82  
4        15.75     11.01      3.28         2.96         33.00  
5        11.27      8.89     10.22         1.00         31.37

Top 10 NES games

In [44]:
top_sellers(cat='Platform',val='NES',n=10)

Name Platform    Year      Genre Publisher  \
Rank                                                                          
2                   Super Mario Bros.      NES  1985.0   Platform  Nintendo   
10                          Duck Hunt      NES  1984.0    Shooter  Nintendo   
23                Super Mario Bros. 3      NES  1988.0   Platform  Nintendo   
97                Super Mario Bros. 2      NES  1988.0   Platform  Nintendo   
128               The Legend of Zelda      NES  1986.0     Action  Nintendo   
156                            Tetris      NES  1988.0     Puzzle  Nintendo   
216                         Dr. Mario      NES  1990.0     Puzzle  Nintendo   
252   Zelda II: The Adventure of Link      NES  1987.0  Adventure  Nintendo   
277      Teenage Mutant Ninja Turtles      NES  1989.0     Action    Palcom   
278                        Excitebike      NES  1984.0     Racing  Nintendo   

      NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                           
2        29.08      3.58      6.81         0.77         40.24  
10       26.93      0.63      0.28         0.47         28.31  
23        9.54      3.44      3.84         0.46         17.28  
97        5.39      1.18      0.70         0.19          7.46  
128       3.74      0.93      1.69         0.14          6.51  
156       2.97      0.69      1.81         0.11          5.58  
216       2.62      0.60      1.52         0.10          4.85  
252       2.19      0.50      1.61         0.08          4.38  
277       3.38      0.44      0.31         0.04          4.17  
278       2.04      0.48      1.57         0.07          4.16

Top 10 selling games in the year 2000

In [45]:
top_sellers(cat='Year',val=2000,n=10)

Name Platform    Year         Genre  \
Rank                                                                      
133              Pokémon Crystal Version       GB  2000.0  Role-Playing   
174                     Final Fantasy IX       PS  2000.0  Role-Playing   
224                             Driver 2       PS  2000.0        Action   
226             Tony Hawk's Pro Skater 2       PS  2000.0        Sports   
243   Dragon Quest VII: Warriors of Eden       PS  2000.0  Role-Playing   
295                Tekken Tag Tournament      PS2  2000.0      Fighting   
334            Spyro: Year of the Dragon       PS  2000.0      Platform   
359                       WWF SmackDown!       PS  2000.0      Fighting   
368          Rugrats in Paris: The Movie       PS  2000.0        Action   
394                           Crash Bash       PS  2000.0          Misc   

                        Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
Rank                                                                           
133                      Nintendo      2.55      1.56      1.29         0.99   
174                    SquareSoft      1.62      0.77      2.78         0.14   
224                         Atari      2.36      2.10      0.02         0.25   
226                    Activision      3.05      1.41      0.02         0.20   
243              Enix Corporation      0.20      0.14      4.10         0.02   
295            Namco Bandai Games      1.68      1.51      0.51         0.35   
334   Sony Computer Entertainment      1.93      1.58      0.00         0.19   
359                           THQ      2.01      1.35      0.06         0.16   
368                           THQ      1.96      1.33      0.00         0.23   
394   Sony Computer Entertainment      1.56      1.47      0.19         0.17   

      Global_Sales  
Rank                
133           6.39  
174           5.30  
224           4.73  
226           4.68  
243           4.47  
295           4.05  
334           3.71  
359           3.58  
368           3.52  
394           3.39

Top 10 selling sports games 

In [46]:
top_sellers(cat='Genre',val='Sports',n=10)

Name Platform    Year   Genre  \
Rank                                                                
1                             Wii Sports      Wii  2006.0  Sports   
4                      Wii Sports Resort      Wii  2009.0  Sports   
14                               Wii Fit      Wii  2007.0  Sports   
15                          Wii Fit Plus      Wii  2009.0  Sports   
78                               FIFA 16      PS4  2015.0  Sports   
86    Mario & Sonic at the Olympic Games      Wii  2007.0  Sports   
113                              FIFA 14      PS3  2013.0  Sports   
118                        Zumba Fitness      Wii  2010.0  Sports   
122                              FIFA 12      PS3  2011.0  Sports   
125                              FIFA 15      PS4  2014.0  Sports   

            Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
Rank                                                                            
1            Nintendo     41.49     29.02      3.77         8.46         82.74  
4            Nintendo     15.75     11.01      3.28         2.96         33.00  
14           Nintendo      8.94      8.03      3.60         2.15         22.72  
15           Nintendo      9.09      8.59      2.53         1.79         22.00  
78    Electronic Arts      1.11      6.06      0.06         1.26          8.49  
86               Sega      2.58      3.90      0.66         0.91          8.06  
113   Electronic Arts      0.78      4.32      0.07         1.73          6.90  
118         505 Games      3.50      2.64      0.00         0.67          6.81  
122   Electronic Arts      0.84      4.32      0.11         1.42          6.69  
125   Electronic Arts      0.79      4.29      0.05         1.47          6.59

### Exercise #3

Generate a (linear) correlation matrix between sales in specific markets (North America, Europe, Japan, Other) and global sales. 

In [63]:
data[['NA_Sales','EU_Sales','JP_Sales','Other_Sales','Global_Sales']].corr()

NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales
NA_Sales      1.000000  0.767727  0.449787     0.634737      0.941047
EU_Sales      0.767727  1.000000  0.435584     0.726385      0.902836
JP_Sales      0.449787  0.435584  1.000000     0.290186      0.611816
Other_Sales   0.634737  0.726385  0.290186     1.000000      0.748331
Global_Sales  0.941047  0.902836  0.611816     0.748331      1.000000

What do you observe about the relationship between sales in these various markets?

North America Sales & Europe Sales highly correlate with a positive 0.76. There is also high correlation between Europe and Other Sales with 0.72. North America also correlates well with Other Sales with 0.63. Japan does have positive correlation with North America, Europe and Other but with very small values of 0.44, 0.43 and 0.29 respectively.

North America and Europe Sales correlate very highly with global sales followed by Other and Japan Sales but with comparatively lesser values.

Then, repeat this calculation for the subset of video games released in the year 2000, and compare these results to those for the subset of video games released in the year 2015.

In [9]:
data[data['Year'] == 2000][['NA_Sales','EU_Sales','JP_Sales','Other_Sales','Global_Sales']].corr()

NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales
NA_Sales      1.000000  0.866555  0.139115     0.691317      0.900815
EU_Sales      0.866555  1.000000  0.108401     0.748552      0.866616
JP_Sales      0.139115  0.108401  1.000000     0.226852      0.512446
Other_Sales   0.691317  0.748552  0.226852     1.000000      0.761891
Global_Sales  0.900815  0.866616  0.512446     0.761891      1.000000

In [75]:
data[data['Year'] == 2015][['NA_Sales','EU_Sales','JP_Sales','Other_Sales','Global_Sales']].corr()

NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales
NA_Sales      1.000000  0.736844  0.123217     0.866672      0.911555
EU_Sales      0.736844  1.000000  0.132160     0.926506      0.922387
JP_Sales      0.123217  0.132160  1.000000     0.126803      0.306981
Other_Sales   0.866672  0.926506  0.126803     1.000000      0.956540
Global_Sales  0.911555  0.922387  0.306981     0.956540      1.000000

What do you observe about how the relationship between the sales figures has changed over the 15-year period?

In year 2000, North America had the highest correlation with Global Sales followed by Europe, Other and Japan. But in 2015, the order changed to Other, Europe, North America and Japan.

Also in this period of 15 years, correlation between North America and Europe dropped a bit from 0.86 to 0.73. North American, European & Other Sales continue to be uncorrelated with Japanese Sales. But the correlation between North America and Other Sales increased.